app/(moderation)/dashboard/profile/[id]/page.tsx

```diff
@@ -0,0 +1,32 @@
+import UserProfileHeader from '@/components/Headers/UserProfileHeader/UserProfileHeader'
+import UserProfileDetails from '@/components/UserProfile/UserProfileDetails/UserProfileDetails'
+import UserProfileMain from '@/components/UserProfile/UserProfileMain/UserProfileMain'
+import ModerationActionHeader from '@/components/Headers/ModerationActionHeader/ModerationActionHeader'
+import { AppRoutes } from '@/utils/routes'
+import { redirect } from 'next/navigation'
+import { Role } from '@prisma/client'
+import { requireUserRoles } from '@/utils/auths'
+import { getProfileByUserId } from '@/backend/profile/profile.service'
+
+import styles from './page.module.scss'
+
+export default async function ModerationUserProfile({
+  params,
+}: {
+  params: { id: string }
+}) {
+  const profile = await getProfileByUserId(params.id)
+
+  if (!profile || !requireUserRoles([Role.MODERATOR]))
+    redirect(AppRoutes.dashboard)
+
```

Comment:
Should the user be redirected to AppRoutes.dashboard if they're not a moderator? It seems this condition will allow non-moderator access to the dashboard. Is this intended?